test_common.py

```diff
@@ -0,0 +1,219 @@
+    for name in NON_SYMMETRIC_METRICS:
+        metric=ALL_METRICS[name]
+        assert_almost_equal(metric([0,1,2,5,4,9],[0,1,9,4,3,5]),
+                            metric([0,1,9,4,3,5],[0,1,2,5,4,9]),
+                            err_msg="%s is not symmetric" % name)
+                            
+      
+                    
+#test function for metrics whose output in range 0 to 1
+#bicluster metrics have different input format so they are handled differently                  
+def test_normalized_output_bicluster():
+    # test for jaccard
+    a1 = np.array([True, True, False, False])
+    a2 = np.array([True, True, True, True])
+    a3 = np.array([False, True, True, False])
+    a4 = np.array([False, False, True, True])
+    assert_equal(_jaccard(a1, a1, a1, a1), 1)
+    assert_greater(_jaccard(a1, a1, a2, a2), 0)
+    assert_less(_jaccard(a1, a1, a2, a2), 1)
+    assert_greater(_jaccard(a1, a1, a3, a3),0)
+    assert_less(_jaccard(a1, a1, a3, a3), 1)
+    assert_equal(_jaccard(a1, a1, a4, a4), 0)
+    
+    #test for consensus score 
+    a = [[True, True, False, False],
+        [False, False, True, True]]
+    b = a[::-1]
+    assert_equal(consensus_score((a, a), (a, a)), 1)
+    assert_equal(consensus_score((a, a), (b, b)), 1)
+    assert_equal(consensus_score((a, b), (a, b)), 1)
+    assert_equal(consensus_score((a, b), (b, a)), 1)
+
+    assert_equal(consensus_score((a, a), (b, a)), 0)
+    assert_equal(consensus_score((a, a), (a, b)), 0)
+    assert_equal(consensus_score((b, b), (a, b)), 0)
+    assert_equal(consensus_score((b, b), (b, a)), 0)
+    
+    b=a[: 2]
+    assert_greater(consensus_score((a , a), (b , b)), 0)
+    assert_equal(consensus_score((a , a), (b , b)), 1)
+        
+#test function for metrics whose output in range 0 to 1
+def test_normalized_output():
+    for name in METRICS_NORMALIZED_OUTPUT:
+        metric=ALL_METRICS[name]
+        assert_greater(metric([0, 0, 0, 1, 1, 1],[0, 0, 0, 1, 2, 2]), 0.0)
+        assert_less(metric([0, 0, 0, 1, 1, 1],[0, 1, 0, 1, 2, 2]), 1.0)
+        assert_greater(metric([0, 0, 1, 1, 2, 2],[0, 0, 1, 1, 1, 1]), 0.0)
+        assert_less(metric([0, 0, 0, 2, 2, 2],[0, 1, 0, 1, 2, 2]),1.0)
+        assert_greater(metric([0, 0, 0, 2, 2, 2],[0, 1, 0, 1, 2, 2]), 0.0)
+
+#when information is zero these metrics output zero                     
+def test_exactly_zero_info_score():
+    # Check numerical stability when information is exactly zero
+    for i in np.logspace(1, 4, 4).astype(np.int):
+        labels_a, labels_b = (np.ones(i, dtype=np.int),
+                              np.arange(i, dtype=np.int))
+        assert_equal(normalized_mutual_info_score(labels_a, labels_b), 0.0)
+        assert_equal(v_measure_score(labels_a, labels_b), 0.0)
+        assert_equal(adjusted_mutual_info_score(labels_a, labels_b), 0.0)
+        assert_equal(normalized_mutual_info_score(labels_a, labels_b), 0.0)
+    
+#test function for mtericshaving the property of not changing the score when 

```

Comment:
I think you're confusing what i meant by invariant to label permutation.

I mean that for some (binary) `y_true, y_pred`, `metric(y_true, y_pred) == metric(1 - y_true, y_pred) == metric(y_true, 1 - y_pred) == metric(1-y_true, 1-y_pred)`